python/mpt/src/mpt/trie_diff.py

```diff
@@ -153,30 +144,31 @@ def resolve(node: Hash32 | Bytes | None | InternalNode) -> InternalNode | None:
                     self._compute_diff(
                         None,
                         r_node.subnodes[i],
-                        path + i.to_bytes(1, "big"),
+                        path + bytes([i]),
                         process_leaf_diff,
                     )
 
             case (LeafNode(), None):
                 # deleted leaf (should not happen post-cancun)
-                preimage = nibble_path_to_bytes(path + l_node.rest_of_key)
-                process_leaf_diff(preimage, l_node, None)
+                full_path = nibble_path_to_bytes(path + l_node.rest_of_key)
+                process_leaf_diff(full_path, l_node, None)
 
             case (LeafNode(), LeafNode()):
                 if l_node.value != r_node.value:
-                    preimage = nibble_path_to_bytes(path + l_node.rest_of_key)
-                    process_leaf_diff(preimage, l_node, r_node)
+                    assert (
```

Comment:
makes me wonder if this case is actually possible, in case of a deletion and an insertion inside the same branch node, at the same nibble

pre state - leaf that is a branch subnode at nibble 0, rest of key is abcde
post state - leaf that is a branch subnode at nibble 0, rest of key is deade